packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx

```diff
@@ -0,0 +1,276 @@
+import graphql from 'babel-plugin-relay/macro'
+import React, {useCallback, useEffect} from 'react'
+import {useFragment} from 'react-relay'
+import {ActivityDetailsQuery} from '~/__generated__/ActivityDetailsQuery.graphql'
+import {TemplateDetails_user$key} from '~/__generated__/TemplateDetails_user.graphql'
+import {TemplateDetails_templates$key} from '~/__generated__/TemplateDetails_templates.graphql'
+import {
+  CATEGORY_ID_TO_NAME,
+  CATEGORY_THEMES,
+  CategoryID,
+  QUICK_START_CATEGORY_ID
+} from '../Categories'
+import {IntegrationsList} from './components/IntegrationsList'
+import clsx from 'clsx'
+import {useHistory} from 'react-router'
+import {DetailsBadge} from './components/DetailsBadge'
+import {ContentCopy} from '@mui/icons-material'
+import useAtmosphere from '../../../hooks/useAtmosphere'
+import useModal from '../../../hooks/useModal'
+import useMutationProps from '../../../hooks/useMutationProps'
+import CloneTemplate from '../../../modules/meeting/components/CloneTemplate'
+import {UnstyledTemplateSharing} from '../../../modules/meeting/components/TemplateSharing'
+import RemovePokerTemplateMutation from '../../../mutations/RemovePokerTemplateMutation'
+import RemoveReflectTemplateMutation from '../../../mutations/RemoveReflectTemplateMutation'
+import {setActiveTemplate} from '../../../utils/relay/setActiveTemplate'
+import useTemplateDescription from '../../../utils/useTemplateDescription'
+import DetailAction from '../../DetailAction'
+import FlatButton from '../../FlatButton'
+import PokerTemplateScaleDetails from '../../../modules/meeting/components/PokerTemplateScaleDetails'
+import TeamPickerModal from '../TeamPickerModal'
+import {MEETING_TYPE_DESCRIPTION_LOOKUP, MEETING_TYPE_TIP_LOOKUP} from './hooks/useActivityDetails'
+
+type Template =
+  ActivityDetailsQuery['response']['viewer']['availableTemplates']['edges'][number]['node']
+
+interface Props {
+  selectedTemplate: Template
+  isEditing: boolean
+  setIsEditing: (isEditing: boolean) => void
+  viewerRef: TemplateDetails_user$key
+  templatesRef: TemplateDetails_templates$key
+}
+
+export const TemplateDetails = (props: Props) => {
```

Comment:
This pretty much untouched and could be a subject to further refactoring. For now, it was just taken out from the main component, to allow displaing other UI needed for meetings.